[BUG] `install` and `cache add` disagrees on accept header

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Offline installs fails with `ENOTCACHED` after caching dependencies using `npm cache add <dep>`:

```
% npm --cache /tmp/cache cache add nan
npm notice 
npm notice New minor version of npm available! 10.5.1 -> 10.7.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.7.0
npm notice Run npm install -g npm@10.7.0 to update!
npm notice 

% npm --cache /tmp/cache --offline install nan
npm ERR! code ENOTCACHED
npm ERR! request to https://registry.npmjs.org/nan failed: cache mode is 'only-if-cached' but no cached response is available.

npm ERR! A complete log of this run can be found in: /tmp/cache/_logs/2024-05-03T09_19_30_468Z-debug-0.log
```

### Expected Behavior

```
% npm --cache /tmp/cache cache add nan
% npm --cache /tmp/cache --offline install nan

added 1 package, and audited 2 packages in 261ms

found 0 vulnerabilities
```

### Steps To Reproduce

As above:
```
npm --cache /tmp/cache add nan
npn --cache /tmp/cache --offline install nan
```

### Findings

By comparing the output of
```js
cacache.ls("/tmp/cache/_cacache").then(console.log)
```
from populating the cache with first `npm install` and then with `npm cache add` I found that the `accept:` header is different:

`npm install`: `accept: 'application/json'`
`npm cache add`: `accept: 'application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*'`

### Environment

- npm: 10.5.1
- Node.js: v21.7.2
- OS Name: Arch Linux
- System Model Name: Lenovo T14s
- npm config:
```ini
; node bin location = /usr/bin/node
; node version = v21.7.2
; npm local prefix = /tmp/scratch
; npm version = 10.5.1
; cwd = /tmp/scratch
; HOME = /nas/home/martin
; Run `npm config ls -l` to show all defaults.
```

--- a/lib/commands/cache.js
+++ b/lib/commands/cache.js
@@ -152,15 +152,20 @@ class Cache extends BaseCommand {
       throw this.usageError('First argument to `add` is required')
     }
 
-    return Promise.all(args.map(spec => {
+    await Promise.all(args.map(async spec => {
       log.silly('cache add', 'spec', spec)
       // we ask pacote for the thing, and then just throw the data
       // away so that it tee-pipes it into the cache like it does
       // for a normal request.
-      return pacote.tarball.stream(spec, stream => {
+      await pacote.tarball.stream(spec, stream => {
         stream.resume()
         return stream.promise()
       }, { ...this.npm.flatOptions })
+
+      await pacote.manifest(spec, {
+        ...this.npm.flatOptions,
+        fullMetadata: true,
+      })
     }))
   }
 

--- a/test/lib/commands/cache.js
+++ b/test/lib/commands/cache.js
@@ -71,7 +71,11 @@ t.test('cache add single pkg', async t => {
     registry: npm.config.get('registry'),
   })
   const manifest = registry.manifest({ name: pkg })
-  await registry.package({ manifest, tarballs: { '1.0.0': path.join(npm.prefix, 'package') } })
+  await registry.package({
+    manifest,
+    times: 2,
+    tarballs: { '1.0.0': path.join(npm.prefix, 'package') },
+  })
   await npm.exec('cache', ['add', pkg])
   t.equal(joinedOutput(), '')
   // eslint-disable-next-line max-len
@@ -99,9 +103,13 @@ t.test('cache add multiple pkgs', async t => {
   })
   const manifest = registry.manifest({ name: pkg })
   const manifest2 = registry.manifest({ name: pkg2 })
-  await registry.package({ manifest, tarballs: { '1.0.0': path.join(npm.prefix, 'package') } })
   await registry.package({
-    manifest: manifest2, tarballs: { '1.0.0': path.join(npm.prefix, 'package') },
+    manifest,
+    times: 2,
+    tarballs: { '1.0.0': path.join(npm.prefix, 'package') },
+  })
+  await registry.package({
+    manifest: manifest2, times: 2, tarballs: { '1.0.0': path.join(npm.prefix, 'package') },
   })
   await npm.exec('cache', ['add', pkg, pkg2])
   t.equal(joinedOutput(), '')

